outport/firehose/firehoseIndexer.go

```diff
@@ -122,6 +131,57 @@ func (fi *firehoseIndexer) getHeaderBytes(headerHandler data.HeaderHandler) ([]b
 	return headerBytes, headerType, err
 }
 
+func getBody(bodyHandler data.BodyHandler) (*block.Body, error) {
+	if check.IfNil(bodyHandler) {
+		return &block.Body{}, nil
```

Comment:
So if bodyHandler is nil we will return an empty block body ?

Reply:
Refactored the code to return `nil`, `errNilBlockBody` in case of nil block body.
In the caller code, however, I check if the error is `errNilBlockBody` and, if so, I fill the body field directly with nil